src/dev_deps.ts

```diff
@@ -8,5 +8,24 @@ export {
 export * as mock from "https://deno.land/std@0.152.0/testing/mock.ts";
 
 export { assert } from "https://deno.land/x/conditional_type_checks@1.0.6/mod.ts";
+export type {
+  IsAny,
+  IsExact,
+} from "https://deno.land/x/conditional_type_checks@1.0.6/mod.ts";
 
-export type { IsExact } from "https://deno.land/x/conditional_type_checks@1.0.6/mod.ts";
+/**
+ * Checks whether T includes U.
+ */
+export type CanBe<T, U> = Extract<T, U> extends never ? false : true;
+/**
+ * Checks whether T can never include U.
+ */
+export type CannotBe<T, U> = Extract<T, U> extends never ? true : false;
+/**
+ * Checks whether the provided type parameter allows to be undefined.
+ * Useful for checking optionality.
+ */
+export type CanBeUndefined<T> = CanBe<T, undefined> extends true ? true
+  : false;
+export type CannotBeUndefined<T> = CannotBe<T, undefined> extends true ? true
+  : false;
```

Comment:
I'm thinking we might want to include these in our `src/test_utils.ts` folder instead

Reply:
It's been a while - what's your opinion on this now @shapirone ?

Reply:
I feel the same, `dev_deps` is intended for importing so AFAIK we shouldn't be defining our own things in here 😅